azure/azure.go

```diff
@@ -46,11 +59,101 @@ var (
 
 type azureOps struct {
 	cloudops.Compute
-	instance          string
-	resourceGroupName string
-	disksClient       *compute.DisksClient
-	vmsClient         vmsClient
-	snapshotsClient   *compute.SnapshotsClient
+	instance           string
+	resourceGroupName  string
+	managedClusterName string
+	agentPoolName      string
+	disksClient        *compute.DisksClient
+	vmsClient          vmsClient
+	snapshotsClient    *compute.SnapshotsClient
+	agentPoolsClient   *containerservice.AgentPoolsClient
+}
+
+// NewAzureClient initializes cloudops driver for azure based on environment
+// variables or based on instance metadata info available inside Azure VM
+func NewAzureClient() (cloudops.Ops, error) {
```

Comment:
NewAzureClient does not tell what is different in this method than the other two - NewClient and NewEnvClient.
May be call it NewClientFromMetadata?

Reply:
Renamed function